From e79ee7d7fdac204b28d0ea526d84a7213357b293 Mon Sep 17 00:00:00 2001
From: Ashwin Adiga <ashwin.adiga@infineon.com>
Date: Thu, 27 Apr 2023 03:44:56 -0500
Subject: [PATCH 195/208] brcmfmac: fix for guard interval for iw set bitrates

Guard Interval was not getting set properly with cmds:
iw dev wlan0 set bitrates he-mcs-2.4 1:1 he-gi-2.4 0.8
iw dev wlan0 set bitrates he-mcs-5 1:1 he-gi-5 1.6
iw dev wlan0 set bitrates he-mcs-6 1:1 he-gi-6 3.2


Signed-off-by: Ashwin Adiga <ashwin.adiga@infineon.com>
---
 drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
index c680f2d5a2ee..9f1747b3f9a4 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
@@ -6743,7 +6743,7 @@ brcmf_cfg80211_set_bitrate(struct wiphy *wiphy, struct net_device *ndev,
 		rspec = WL_RSPEC_ENCODE_HE;     /* 11ax HE */
 		rspec |= (WL_RSPEC_HE_NSS_UNSPECIFIED << WL_RSPEC_HE_NSS_SHIFT) | (mcs - 1);
 		/* set the other rspec fields */
-		hegi = mask->control[band].he_gi;
+		hegi = mask->control[band].he_gi + 1;
 		rspec |= ((hegi != 0xFF) ? HE_GI_TO_RSPEC(hegi) : 0);
 
 		if (band == NL80211_BAND_2GHZ)
-- 
2.17.1

